src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  static DAMAGE = 50;
+
+  bite(prey) {
+    if (prey instanceof Carnivore) {
+      return;
+    }
+
+    if (prey.hidden === true) {
+      return;
+    }
+
+    if (prey.health <= Carnivore.DAMAGE) {
+      Animal.alive.splice(
+        Animal.alive.findIndex((animal) => animal === prey),
+        1,
+      );
+
+      return;
+    }
```

Comment:
The logic for removing the prey from the `Animal.alive` array is correct, but it could be simplified. Consider using `Animal.alive = Animal.alive.filter(animal => animal !== prey);` to make the code more concise and readable.